Reword and add uniformity to Standard Library documentation (#4464)
edgedb/edgedb: docs/stdlib/objects.rst

      - Root for user-defined object types

> :eql:type:`BaseObject` is the root of the object type hierarchy and all
> object types in EdgeDB, including system types, whether extending it directly
> or indirectly. User-defined object types extend from the :eql:type:`Object`
> type, which is a subtype of ``std::BaseObject``.

Reviewer: ```suggestion
:eql:type:`BaseObject` is the root of the object type hierarchy and all object
types in EdgeDB, including system types, extend it either directly or
indirectly. User-defined object types extend from the :eql:type:`Object` type,
which is a subtype of ``std::BaseObject``.
```

---------

.. eql:type:: std::BaseObject

>     Represents the root object of a type:

Reviewer: ```suggestion
    The root object type.
```
The rewrite changes the meaning. ``BaseObject`` doesn't represent anything — it is a type — and it's not the root object _of_ a type — it is the root object type, meaning the root type of all objects.

            required readonly link __type__ -> schema::ObjectType;
        }

>     Some subtypes may override the ``id`` property, but only to validate
>     UUID generation functions. Currently, these include
>     :eql:func:`uuid_generate_v1mc` and :eql:func:`uuid_generate_v4`.

Reviewer: ```suggestion
    Subtypes may override the ``id`` property, but only with a valid UUID
    generation function. Currently, these are
    :eql:func:`uuid_generate_v1mc` and :eql:func:`uuid_generate_v4`.
```

I recommend mostly reverting here. The rewrite introduces some issues.
- "Some" is already implied by "may" (not all of them do it), so it doesn't add anything here.
- The ``id`` is not overridden for the purpose of validating a UUID generation function but is overridden _to_ a valid UUID generation function. "with" might actually be clearer here than "to," so I'll incorporate that into my suggestion.
- Swapping "are" for "include" implies there may be other valid UUID functions the user can override their ID with, which I don't believe is the case.
